### ISSUE TITLE
Survey Settings Revamp

### ISSUE THREAD

**Issue by sirkotsky** [formbricks/design#13]
# Revamp the survey settings experience

## Problem 
Today, the survey configuration is placed under the "Audience" tab. It is far from being intuitive, and the user stumbles upon many challenges. This inevitably impacts the number of surveys being launched successfully, as well as the user experience. 

![Formbricks Survey Creation Flow](https://user-images.githubusercontent.com/22578094/236610320-d2ca389f-fdbf-448e-b8e5-5ae08fba0220.png)

## Hypothesis
> Improving the experience of adding and configuing the survey will help increase the number of surveys created on the platform. 

### Areas for improvement:
![Formbricks Survey Creation Areas For Improvements](https://user-images.githubusercontent.com/22578094/236610758-1e4d3f9a-ba63-45af-8faf-bd7c7da2aa46.png)

1. New survey creation: from [the past studies](https://github.com/formbricks/design/blob/main/insights/23042023_users_survey.md#observations-from-the-past-studies), we know that users struggle to locate the "Create survey" button;
2. Audience tab: [past studies](https://github.com/formbricks/design/blob/main/insights/23042023_users_survey.md#observations-from-the-past-studies) and [UX Audit](https://github.com/formbricks/design/blob/main/insights/20042023_ux_audit.md) confirm it to be confusing and not intuitive;
3. The audience tab content can be further improved as well.

## Impact and success metrics
- Enhanced customer exprience;
- Increased number of surveys created on the platform.

## Timeline
It is a high-priority design job with high impact and relatively moderate scope. Upon closer examination, it might require a few more drastic changes than initially expected.

**Comment by sirkotsky** [comment]
# Early proposal
> [Penpot link](https://design.penpot.app/#/view/248140e0-d822-80a6-8002-7747229eabc1?page-id=248140e0-d822-80a6-8002-7747229eabc2&section=interactions&index=0&share-id=248140e0-d822-80a6-8002-79b87bb4f40e)

![image](https://user-images.githubusercontent.com/22578094/236741255-0077de53-1682-4a40-ab27-b83f210922f3.png)

This is an early proposal. Below is a breakdown of my discovery process, a detailed overview of all changes, and proposed next steps. 
## Market analysis
![image](https://user-images.githubusercontent.com/22578094/236740661-93ca9aba-9bb3-4cbe-90b7-cc4f2580ff66.png)

I reviewed some products I could get my hands on and made some rather interesting observations:
- Qualtrics, SurveyMonkey, Typeform (1, 2, 4) feature (one way or the other) a grid of templates, and open a template preview is a new page;
- Refiner's (3) demo page has a rich demo page with a dropdown, allowing to quickly switch between templates;
- Sprig's (5) template tiles show a preview of a template.

Formbricks is somewhat closer to Refiner in that regard, but allows switching between templates in the left panel. We could retain the existing behaviour (the interaction happening primarily in the left panel) by splitting the page view:

![image](https://user-images.githubusercontent.com/22578094/236767602-22279a7e-c03e-4fc1-8000-2f5cd09203bb.png)

## Concept design
### Overview
After a round of explorations, I've managed to slightly tweak the existing UI and experience and focus on improving them (rather than revamping) to increase conversion and reduce ambiguity. 

### New Survey
Below is the breakdown of a "New Survey" page redesign:

1. We remove the secondary top navigation and change the title. The user will still be able to escape the page by clicking the Formbricks logo, but the focus of the page will remain on the action;
2. We enlarge the area for preview and centre it to bring the user's attention. Moreover, we change the default button style to avoid confusion between Formbricks CTA and preview (as we have today);
3. Finally, once selected, the template will feature a button clearly stating "Use this template". This will allow to avoid confusion and clearly signal the user what they must do to activate the survey.

![image](https://user-images.githubusercontent.com/22578094/236740686-2c73436e-3261-495f-b1d4-680d2cc003b2.png)

Alternatively, we can explore a slightly different approach:

1. Once the template is selected, pull down a sticky bottom menu with the survey name;
2. Place the CTA at the bottom where the user might expect to see it. 

![image](https://user-images.githubusercontent.com/22578094/236740700-82cf8b5f-0160-4d40-9bb1-06aad035e6af.png)

I would, however, prefer using the first concept. It is providing the most obvious CTA placement and retains the focus in the Action area for the user.

### Survey Settings
I have slightly redesigned the Audience page as well, albeit slightly more drastically:

1. The Formbricks top bar is gone, allowing the survey to take more place. Especially relevant for small monitors. Notice how we display the Product Name, and show that the Survey is saved as draft;
2. We also only display 2 buttons: Cancel (alternative is to place "Back" in the left hand corner, with a clear label) and Publish;
3. Audience is renamed to Settings (more consistent with other products, better reflecting the nature and the functionality of the page); 
4. The page itself is revamped as well: from copy change to bringing the clear warning states. For instance, the user cannot publish a survey, and here we explain precisely why;
5. Notice how the visual style of the "Coming soon" banners has changed: from warning (existing) to enticing. The actions are now more clear and all visible upfront (as opposed to being hidden today);
6. Checkbox replaced with a toggle: technically, a more accurate reflection of the "On/Off" action.

![image](https://user-images.githubusercontent.com/22578094/236740718-0ad11e90-f013-41f5-8c33-c8399cf5eeff.png)

Alternatively, we can explore making the page shorter and by introducing dropdowns (1) where applicable:

![image](https://user-images.githubusercontent.com/22578094/236740731-49b76c3f-305b-4844-8230-95e7a19bd3ed.png)

## Next steps
Some of the proposed changes, especially those concerning the UI, could be implemented rather quickly and tested live in the production environment. I would also love to collect some feedback from the community in Discord. 

In general, I would love to hear @jobenjada out, knowing that revamping these pages was the top priority, and see how we can move from here and how we are going to measure the impact. Cheers!

**Comment by jobenjada** [comment]
## Template View Conversion

### Addition to research
**Sprig** also opens a new modal with the Template:
![grafik](https://user-images.githubusercontent.com/72809645/236874017-c40209af-2550-4bef-b0c3-9468be3150b0.png)

![grafik](https://user-images.githubusercontent.com/72809645/236874178-5ee5e621-377c-4696-8f4c-4d2150569a81.png)

They also include text to explain the Template but I highly doubt that a single user has read this:
![grafik](https://user-images.githubusercontent.com/72809645/236874511-669b4d74-b9ce-46a5-a4fa-ba717b78a6c0.png)


Ours looks indeed most like what **Refiner** did - very streamlined. They also display surveys randomly in different format what I find confusing (mobile, top left, bottom right, etc.:
![grafik](https://user-images.githubusercontent.com/72809645/236874345-361f94c8-4780-4e07-ba3d-58711462f254.png)

### Feedback on suggestions re Template View

- I'd like to keep the current setup (action left, preview right).
- Removing secondary top nav: We kept it because in the survey editor the main CTA is also in the top right corner. This is why I also don't like to add another CTA in the bottom right corner
- Bringing it to center: I full see the point that the current way is confusing. The thing is that we display it in the bottom right corner, because the widget will be displayed in the bottom right corner in the users app (which isn't clear). I like the idea of moving it to the center and. A different approach could be to let users preview different positions and channels (link survey vs. in-app vs. email in a mockup). In both ways, positioning it centrally and adding "Survey Preview" to the window while removing the frame is a good call!
- Adding the CTA into the template selection makes a lot of sense - lets do it!

### KPI: Dropoff Rate Template View -> Survey Editor (currently 21% dropoff)

------

## Settings Tab in Editor

1. No NavBar works for me, streamlines it
2. Love the Product Name and Saved as draft, very clear
3. With no Save Button: What do people do who dont want to publish their survey right away? We intentionally decided against auto-save because it could fuck up surveys which are already running in prod in peoples products.
4. Great idea with the warnings!! Really like it and makes it very clear, also in comparison to link surveys.
5. We were afraid that users would get drowned in settings because there is so much to setup. Up next is the attribute filtering which is similiar to the Survey Trigger UI. What do you think?
6. Like the toggle!

Not a big fan of the dropdowns bc. the current UI allows a bit more explanation. 

### KPI: Dropoff Rate Survey Editor -> Survey Summary (currently 63%)

Due to the significantly higher impact I would prioritize this. However, since the changes for the first are rather small, we can do both.

-----

FYI: We are now working in weekly sprints. We can discuss this further this week (and also invite the community) and line it up for next week :)

Thanks for your great work!

**Comment by sirkotsky** [comment]
Thanks, @jobenjada, will have a closer look and share my thoughts!

I also have a quick assumption to share. The dropoff rate is most likely due to Publish button being disabled: people don't know they need to configure the widget. So how about we include this in onboarding as well, right before they get started?

![image](https://user-images.githubusercontent.com/22578094/236960449-adb657a0-127c-4983-aefa-dbf9e3fbd5af.png)

That, and if Formbricks is not connected yet, the default version should be "Standalone"

![image](https://user-images.githubusercontent.com/22578094/236960579-8b92d4ba-8de9-4c48-937f-0f78dc614b47.png)

This way, we kill 2 birds with one stone: people can still create a survey, they are not blocked by default, and we tell them how it is done, and we do it in an intuitive way.

**Comment by sirkotsky** [comment]
<img width="540" alt="Screenshot 2023-05-09 at 9 14 18 AM" src="https://user-images.githubusercontent.com/22578094/236969383-5d3b24b2-25f8-4a11-b786-7794f977243b.png">

Just a quick draft (sorry, putting it together on the go): we need to let the user know that there's something wrong that's blocking them from launching a survey. We can use a visual signal to warn them that something's wrong with the settings. 

Cheers, will be back with more soon!

**Comment by jobenjada** [comment]
I saw this research Onboardings a few weeks back and quite liked it, because it's helpful and adds a growth loop
![grafik](https://user-images.githubusercontent.com/72809645/237022646-3469828a-932a-4c7c-ae53-7516a13bee7d.png)

Generally, making them aware the to get the full benefit out of Formbricks is required to setup the widget in the Onboarding makes a lot of sense to me.

It kind of goes hand-in-hand with revamping the setup checklist which is currently not very intuitive. We could rework it as well as it's mostly a bit of UI which needs to be updated. I can put something together in the second half of the week to get your thoughts on and we can see if we release it in one go or if its out of scope for now.

Really like the idea of defaulting link surveys when the widget is not implemented - super easy to do!

**Comment by sirkotsky** [comment]
Thanks for bringing it up, @jobenjada, and for sharing the market research! Below is my proposal:

# Proposal
## Onboarding
At the final step of onboarding, we can invite the user to configure their Formbricks widget and help visualise the survey built into the Web App. 

This will help demonstrate a clear value of configuring the widget, but still allow the user to skip the step as optional. 
![image](https://github.com/formbricks/design/assets/22578094/1f00d8d1-9b66-4a2f-8eaf-f3300d3956ae)

## Survey Templates
We can try bringing the same pattern back and using a "browser window" to highlight and contextualise the preview. 
![image](https://github.com/formbricks/design/assets/22578094/edda3186-2e8c-496d-95a2-0aa3af5876c6)

## Survey Settings
In the survey settings, we can implement two fundamental changes:
- Select "Standalone" by default;
- Add a warning to the Web App survey with a link to configuring the widget;

I also brought back the "Back" (it's not styled, so we can retain the existing version with a "Back" arrow instead") and "Save" buttons as per your suggestion.
![image](https://github.com/formbricks/design/assets/22578094/e1af243d-2462-4bbc-a7ee-5b2aa4d9e2d5)

## Success criteria
Aligned on measuring the drop-off rate.

## Next steps
We will need to not only implement the changes, but make sure (as you brought up) the Settings Page and flexible and scalable. This might require us to redesign it further, compress is visually and explore other ways of presenting the elements. For instance, we can reduce paddings and visual clutter, introduce a secondary navigation, add dropdowns and lists where applicable.

Otherwise, looking forward to hearing your comments & tracking the drop-off rate across all pages and act accordingly.

Cheers!

**Comment by jobenjada** [comment]
Hey! 

Looks pretty good :) Im sure a few details will change as I write the tickets but the concept seems solid to me!

A last suggestion: We could think about allowing users to toggle between the survey type in the preview (both and template view and in the survey editor). We currently have the "Update brand color" appear when users hover over the left half of the screen.  If we instead offer a toggle, users might get the difference intuitively without having to read much  text (which no one does).

What do you think?

Apart from that I'll write this up for next weeks sprint :) Thanks a lot!!! 🙏

**Comment by jobenjada** [comment]
It wouldnt be hard to do as we have the logic in the preview already. Its the same toggle we're optimizing on the settings page but with a different frontend.

Id love to hear your considerations :) We wont start working on it before Monday, so there should be time to think it through 🙌

**Comment by sirkotsky** [comment]
I understand that building it won't be a problem from the technical standpoint: for the user, however, having a toggle or a setting anywhere outside the "Settings" panel might be confusing.

If we agree that the right-hand panel is for the "Preview", I would avoid placing any actionable items there (including the brand colour change, by the way). 

We could have some sort of switcher there, but it doesn't quite make sense:
- By default, users can only create a standalone survey without installing the widget,
- This is not a feature that they will need to constantly "toggle", but rather, configure once per survey. 

![image](https://github.com/formbricks/design/assets/22578094/167cef2c-e6d5-4883-a94a-b948de7d85c7)

It could be something to park for the future explorations, especially when more survey types are available, but I'm not sure "Preview" is the right place for anything settings-related.

Cheers!

**Comment by mattinannt** [comment]
also currently (and maybe also in the future) the complete right side (preview) is hidden on mobile. so placing every setting on the left side would be the preferred option for me.

**Comment by jobenjada** [comment]
looks good, plan is to develop it this week and release it on Friday.

**Comment by sirkotsky** [comment]
I'll close the issue now and start compiling the case study! Thank you, folks!
